src/main/kotlin/com/criptext/mail/scenes/signin/data/SignInAPIClient.kt

```diff
@@ -36,4 +38,36 @@ class SignInAPIClient(private val httpClient: HttpClient) {
         return httpClient.getFileStream(path = "/userdata", authToken = token, params = params)
     }
 
+    fun postLinkBegin(recipientId: String): String{
+        val jsonPut = JSONObject()
+        jsonPut.put("targetUsername", recipientId)
+
+        return httpClient.post(path = "/link/begin", authToken = null, body = jsonPut)
+    }
+
+    fun getLinkStatus(jwt: String): String{
+        return httpClient.get(path = "/link/status", authToken = jwt)
+    }
+
+    fun postLinkAuth(recipientId: String, jwt: String): String{
+        val jsonPut = JSONObject()
+        jsonPut.put("recipientId", recipientId)
+        jsonPut.put("deviceName", DeviceUtils.getDeviceName())
```

Comment:
API methods should not know anything about the device. Device data should be obtained elsewehere and passed as parameter.